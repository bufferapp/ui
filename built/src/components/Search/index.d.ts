export { default } from './Search';
//# sourceMappingURL=index.d.ts.map