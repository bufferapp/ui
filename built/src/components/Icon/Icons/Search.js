import React from 'react';
import createIconComponent from '../utils/createIconComponent';
const SearchIcon = createIconComponent({
    content: (React.createElement("g", null,
        React.createElement("path", { fillRule: "evenodd", d: "M9.5 10a3.5 3.5 0 100-7 3.5 3.5 0 000 7zm0 2a5.5 5.5 0 100-11 5.5 5.5 0 000 11z", clipRule: "evenodd" }),
        React.createElement("path", { d: "M5.95 9l1.414 1.414-4.243 4.243a1 1 0 01-1.414-1.414L5.95 9z" }))),
});
SearchIcon.displayName = 'SearchIcon';
export default SearchIcon;
//# sourceMappingURL=Search.js.map