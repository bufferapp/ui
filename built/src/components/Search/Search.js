import React from 'react';
import PropTypes from 'prop-types';
import { SearchInput, SearchWrapper } from './style';
export default class Search extends React.Component {
    constructor() {
        super(...arguments);
        this.state = {
            search: '',
        };
        this.updateSearch = (search) => {
            this.setState({
                search,
            });
        };
        this.onChange = (event) => {
            const { onChange } = this.props;
            const search = event.target.value;
            onChange(search);
            this.updateSearch(search);
        };
        this.clearSearch = () => {
            const { onChange } = this.props;
            onChange('');
            this.updateSearch('');
        };
    }
    componentDidUpdate(prevProps) {
        if (prevProps.isOpen !== this.props.isOpen) {
            setTimeout(() => this.inputRef.focus(), 50);
        }
    }
    render() {
        const { placeholder, onClick, height, clearSearchOnBlur, clearSearchOnFocus, } = this.props;
        const { search } = this.state;
        return (React.createElement(SearchWrapper, null,
            React.createElement(SearchInput, { placeholder: placeholder, type: "text", value: search, ref: (inputRef) => (this.inputRef = inputRef), onChange: (event) => this.onChange(event), onClick: onClick, onBlur: clearSearchOnBlur ? this.clearSearch : undefined, onFocus: clearSearchOnFocus ? this.clearSearch : undefined, height: height })));
    }
}
Search.propTypes = {
    placeholder: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    isOpen: PropTypes.bool,
    onClick: PropTypes.func,
    height: PropTypes.string,
    clearSearchOnBlur: PropTypes.bool,
    clearSearchOnFocus: PropTypes.bool,
    onBlur: PropTypes.func,
};
Search.defaultProps = {
    placeholder: '',
    height: 'tall',
    isOpen: false,
    clearSearchOnBlur: false,
    clearSearchOnFocus: false,
    onClick: () => { },
    onBlur: () => { },
};
//# sourceMappingURL=Search.js.map