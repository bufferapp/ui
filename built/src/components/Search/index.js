export { default } from './Search';
//# sourceMappingURL=index.js.map