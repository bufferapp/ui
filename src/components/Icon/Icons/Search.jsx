/** WARNING
 * This file is automatically generated by the `npm run gen:icons` command.
 * ALL changes will be overwritten.
 */
import React from 'react';
import createIconComponent from '../utils/createIconComponent';

const SearchIcon = createIconComponent({
  content: (
    <g>
      <path
        fillRule="evenodd"
        d="M9.5 10a3.5 3.5 0 100-7 3.5 3.5 0 000 7zm0 2a5.5 5.5 0 100-11 5.5 5.5 0 000 11z"
        clipRule="evenodd"
      />
      <path d="M5.95 9l1.414 1.414-4.243 4.243a1 1 0 01-1.414-1.414L5.95 9z" />
    </g>
  ),
});
SearchIcon.displayName = 'SearchIcon';

export default SearchIcon;
