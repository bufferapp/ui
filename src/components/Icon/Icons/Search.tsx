/** WARNING
 * This file is automatically generated by the `yarn gen:icons` command.
 * ALL changes will be overwritten.
 */
import React from 'react'
import createIconComponent from '../utils/createIconComponent'

const SearchIcon = createIconComponent({
  content: (
    <g>
      <path
        fillRule="evenodd"
        d="M9.5 10a3.5 3.5 0 100-7 3.5 3.5 0 000 7zm0 2a5.5 5.5 0 100-11 5.5 5.5 0 000 11z"
        clipRule="evenodd"
      />
      <path d="M5.95 9l1.414 1.414-4.243 4.243a1 1 0 01-1.414-1.414L5.95 9z" />
    </g>
  ),
})
// @ts-expect-error TS(2339) FIXME: Property 'displayName' does not exist on type '(pr... Remove this comment to see the full error message
SearchIcon.displayName = 'SearchIcon'

export default SearchIcon
