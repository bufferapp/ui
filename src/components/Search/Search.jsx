import React from 'react'
import PropTypes from 'prop-types'
import { SearchInput, SearchWrapper } from './style'

/** Search input that filters the Select items and adds keyboard navigation */
export default class Search extends React.Component {
  state = {
    search: '',
  }

  componentDidUpdate(prevProps) {
    if (prevProps.isOpen !== this.props.isOpen) {
      setTimeout(() => this.inputRef.focus(), 50)
    }
  }

  updateSearch = (search) => {
    this.setState({
      search,
    })
  }

  onChange = (event) => {
    const { onChange } = this.props
    const search = event.target.value

    onChange(search)
    this.updateSearch(search)
  }

  clearSearch = () => {
    const { onChange } = this.props

    onChange('')
    this.updateSearch('')
  }

  render() {
    const {
      placeholder,
      onClick,
      height,
      clearSearchOnBlur,
      clearSearchOnFocus,
    } = this.props

    const { search } = this.state

    return (
      <SearchWrapper>
        <SearchInput
          placeholder={placeholder}
          type="text"
          value={search}
          ref={(inputRef) => (this.inputRef = inputRef)}
          onChange={(event) => this.onChange(event)}
          onClick={onClick}
          onBlur={clearSearchOnBlur ? this.clearSearch : undefined}
          onFocus={clearSearchOnFocus ? this.clearSearch : undefined}
          height={height}
        />
      </SearchWrapper>
    )
  }
}

Search.propTypes = {
  /** Search placeholder */
  placeholder: PropTypes.string,

  /** Function to call on search input change */
  onChange: PropTypes.func.isRequired,

  /** Is the select menu open */
  isOpen: PropTypes.bool,

  /** Is onClick event */
  onClick: PropTypes.func,

  /** Input height */
  height: PropTypes.string,

  /** Should the search clear on blur */
  clearSearchOnBlur: PropTypes.bool,

  /** Should the search clear on focus */
  clearSearchOnFocus: PropTypes.bool,

  /** Is onBlur event */
  onBlur: PropTypes.func,
}

Search.defaultProps = {
  placeholder: '',
  height: 'tall',
  isOpen: false,
  clearSearchOnBlur: false,
  clearSearchOnFocus: false,
  onClick: () => {},
  onBlur: () => {},
}
