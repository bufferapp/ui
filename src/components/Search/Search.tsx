import React from 'react'
import PropTypes from 'prop-types'
import { SearchInput, SearchWrapper } from './style'

/** Search input that filters the Select items and adds keyboard navigation */
export default class Search extends React.Component {
  state = {
    search: '',
  }

  // @ts-expect-error TS(7006) FIXME: Parameter 'prevProps' implicitly has an 'any' type... Remove this comment to see the full error message
  componentDidUpdate(prevProps) {
    // @ts-expect-error TS(2339) FIXME: Property 'isOpen' does not exist on type 'Readonly... Remove this comment to see the full error message
    if (prevProps.isOpen !== this.props.isOpen) {
      // @ts-expect-error TS(2339) FIXME: Property 'inputRef' does not exist on type 'Search... Remove this comment to see the full error message
      setTimeout(() => this.inputRef.focus(), 50)
    }
  }

  // @ts-expect-error TS(7006) FIXME: Parameter 'search' implicitly has an 'any' type.
  updateSearch = (search) => {
    this.setState({
      search,
    })
  }

  // @ts-expect-error TS(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
  onChange = (event) => {
    // @ts-expect-error TS(2339) FIXME: Property 'onChange' does not exist on type 'Readon... Remove this comment to see the full error message
    const { onChange } = this.props
    const search = event.target.value

    onChange(search)
    this.updateSearch(search)
  }

  clearSearch = () => {
    // @ts-expect-error TS(2339) FIXME: Property 'onChange' does not exist on type 'Readon... Remove this comment to see the full error message
    const { onChange } = this.props

    onChange('')
    this.updateSearch('')
  }

  render() {
    const {
      // @ts-expect-error TS(2339) FIXME: Property 'placeholder' does not exist on type 'Rea... Remove this comment to see the full error message
      placeholder,
      // @ts-expect-error TS(2339) FIXME: Property 'onClick' does not exist on type 'Readonl... Remove this comment to see the full error message
      onClick,
      // @ts-expect-error TS(2339) FIXME: Property 'height' does not exist on type 'Readonly... Remove this comment to see the full error message
      height,
      // @ts-expect-error TS(2339) FIXME: Property 'clearSearchOnBlur' does not exist on typ... Remove this comment to see the full error message
      clearSearchOnBlur,
      // @ts-expect-error TS(2339) FIXME: Property 'clearSearchOnFocus' does not exist on ty... Remove this comment to see the full error message
      clearSearchOnFocus,
    } = this.props

    const { search } = this.state

    return (
      <SearchWrapper>
        <SearchInput
          placeholder={placeholder}
          type="text"
          value={search}
          // @ts-expect-error TS(7006) FIXME: Parameter 'inputRef' implicitly has an 'any' type.
          ref={(inputRef) => (this.inputRef = inputRef)}
          // @ts-expect-error TS(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
          onChange={(event) => this.onChange(event)}
          onClick={onClick}
          onBlur={clearSearchOnBlur ? this.clearSearch : undefined}
          onFocus={clearSearchOnFocus ? this.clearSearch : undefined}
          height={height}
        />
      </SearchWrapper>
    )
  }
}

// @ts-expect-error TS(2339) FIXME: Property 'propTypes' does not exist on type 'typeo... Remove this comment to see the full error message
Search.propTypes = {
  /** Search placeholder */
  placeholder: PropTypes.string,

  /** Function to call on search input change */
  onChange: PropTypes.func.isRequired,

  /** Is the select menu open */
  isOpen: PropTypes.bool,

  /** Is onClick event */
  onClick: PropTypes.func,

  /** Input height */
  height: PropTypes.string,

  /** Should the search clear on blur */
  clearSearchOnBlur: PropTypes.bool,

  /** Should the search clear on focus */
  clearSearchOnFocus: PropTypes.bool,

  /** Is onBlur event */
  onBlur: PropTypes.func,
}

// @ts-expect-error TS(2339) FIXME: Property 'defaultProps' does not exist on type 'ty... Remove this comment to see the full error message
Search.defaultProps = {
  placeholder: '',
  height: 'tall',
  isOpen: false,
  clearSearchOnBlur: false,
  clearSearchOnFocus: false,
  onClick: () => {},
  onBlur: () => {},
}
