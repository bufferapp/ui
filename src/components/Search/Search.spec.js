import snap from 'jest-auto-snapshots'
import Search from './Search'

snap(Search, './Search.jsx')
