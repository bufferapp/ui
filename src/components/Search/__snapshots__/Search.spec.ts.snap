// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Search Matches snapshot when boolean prop "clearSearchOnBlur" is set to: "false" 1`] = `
<styled.div>
  <styled.input
    height="jest-auto-snapshots String Fixture"
    onChange={[Function]}
    onClick={[MockFunction]}
    onFocus={[Function]}
    placeholder="jest-auto-snapshots String Fixture"
    type="text"
    value=""
  />
</styled.div>
`;

exports[`jest-auto-snapshots > Search Matches snapshot when boolean prop "clearSearchOnFocus" is set to: "false" 1`] = `
<styled.div>
  <styled.input
    height="jest-auto-snapshots String Fixture"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[MockFunction]}
    placeholder="jest-auto-snapshots String Fixture"
    type="text"
    value=""
  />
</styled.div>
`;

exports[`jest-auto-snapshots > Search Matches snapshot when boolean prop "isOpen" is set to: "false" 1`] = `
<styled.div>
  <styled.input
    height="jest-auto-snapshots String Fixture"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[MockFunction]}
    onFocus={[Function]}
    placeholder="jest-auto-snapshots String Fixture"
    type="text"
    value=""
  />
</styled.div>
`;

exports[`jest-auto-snapshots > Search Matches snapshot when passed all props 1`] = `
<styled.div>
  <styled.input
    height="jest-auto-snapshots String Fixture"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[MockFunction]}
    onFocus={[Function]}
    placeholder="jest-auto-snapshots String Fixture"
    type="text"
    value=""
  />
</styled.div>
`;

exports[`jest-auto-snapshots > Search Matches snapshot when passed only required props 1`] = `
<styled.div>
  <styled.input
    height="tall"
    onChange={[Function]}
    onClick={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</styled.div>
`;
